sklearn/impute/_base.py

```diff
@@ -596,6 +627,10 @@ def _fit(self, X, y=None):
             raise ValueError("'sparse' has to be a boolean or 'auto'. "
                              "Got {!r} instead.".format(self.sparse))
 
+        if self.precomputed and isinstance(X, type(None)):
```

Comment:
Can `X` be `None` as this point?

Reply:
I added it when I started the issue, at this point it seems absurd as, the mask would be passed internally and has no user intervention. So, I'll remove it.

Thanks.